### PR TITLE
Tunneling bugfix

### PR DESCRIPTION
**Pull request by VakarisZ**
#Fixes
> Fixed tunneling bug where local machine gets set as island.

### CODE REVIEW

File: monkey/infection_monkey/network/network_scanner.py [hunk 1]
```diff
@@ -118,3 +118,6 @@ def _is_any_ip_in_subnet(ip_addresses, subnet_str):
             if NetworkRange.get_range_obj(subnet_str).is_in_range(ip_address):
                 return True
         return False
+
+    def on_island(self, server):
```
Comment (danielguardicore) [inline monkey/infection_monkey/network/network_scanner.py]: Should be either under `monkey/infection_monkey/network/info.py` or `monkey/infection_monkey/utils.py`

File: monkey/infection_monkey/monkey.py [hunk 2]
```diff
@@ -175,8 +175,11 @@ def start(self):
                 if monkey_tunnel:
                     monkey_tunnel.set_tunnel_for_host(machine)
                 if self._default_server:
-                    machine.set_default_server(get_interface_to_target(machine.ip_addr) +
-                                               (':'+self._default_server_port if self._default_server_port else ''))
+                    if self._network.on_island(self._default_server):
```
Comment (danielguardicore) [inline monkey/infection_monkey/monkey.py]: Why does this change things?

Comment (VakarisZ) [reply monkey/infection_monkey/monkey.py]: If monkey is on the island, we want to pass the correct island's IP to the exploited host. If monkey is not on island, we don't want to pass current machine's IP as monkey island.

File: monkey/infection_monkey/network/network_scanner.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/infection_monkey/network/network_scanner.py]: hmm... I know this method has little to do with network scanning, but the fact that _ip_addresses are private suggests that they should only be accessed from within it's class. That way it stays true to the encapsulation principle. Would be cool to hear what @ShayNehmad or @itaymmguardicore thinks about this.

**Pull request merged**